http-server/src/main/java/org/triplea/server/access/ApiKeyGenerator.java

```diff
@@ -0,0 +1,48 @@
+package org.triplea.server.access;
+
+import com.google.common.hash.Hashing;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import lombok.Builder;
+import org.triplea.http.client.ApiKey;
+import org.triplea.lobby.server.db.dao.ApiKeyDao;
+import org.triplea.lobby.server.db.dao.UserJdbiDao;
+
+// TODO: Project#12 This is copy/pasted from lobby sub-project
+@Builder
+public class ApiKeyGenerator implements Supplier<ApiKey> {
+
+  @Nonnull private final ApiKeyDao apiKeyDao;
+  @Nonnull private final UserJdbiDao userDao;
+  @Nonnull private final Supplier<ApiKey> keyMaker;
+
+  /**
+   * Generates a random string value that is extremely difficult to guess or crack. <br>
+   * This key generator is intended to have a pretty fast runtime. <br>
+   * <br>
+   * To generate:<br>
+   * (1) We concatenate a UUID and random number together to get a lot of randomness.<br>
+   * (2) The concatenated random value is hashed to prevent guessing the next value. This makes it
+   * difficult to crack the seed value being used.<br>
+   * (3) Finally, we attach nano epoch to make the API key even more difficult to guess. <br>
+   */
+  public static Supplier<ApiKey> createKeyMaker() {
+    return () -> ApiKey.of(randomHash() + System.nanoTime());
+  }
+
+  private static String randomHash() {
+    return Hashing.sha512()
+        .hashString(UUID.randomUUID() + String.valueOf(Math.random()), StandardCharsets.UTF_8)
+        .toString();
+  }
+
+  @Override
+  public ApiKey get() {
+    final ApiKey key = keyMaker.get();
+    apiKeyDao.storeKey(null, key.getValue());
```

Comment:
This is also a thing I don't get, why is the key always null?

Reply:
`storeKey` expects a username so that it can link the API key back to role authorization. The `null` happens for anonymous users, so in effect the connection for lobby watcher, instead of getting a magic name suffix, instead is authenticated as an anonymous user with no name.

The 'import' of this code is kinda quick and dirty for right now, it'll be more clear where this needs to go when we unify the two `ApiKeyGenerator` classes together, I'm concerned about code churn in the meantime and do not want the two classes to drift too much just yet (until we can ideally delete the original of the copy/paste)